[UnifiedPDF] PDFPluginChoiceAnnotations have incorrect font size.

sammygill opened the pull request:
#### a2247463abb0064a8213ab4d24a23c9d38378514
<pre>
[UnifiedPDF] PDFPluginChoiceAnnotations have incorrect font size.
<a href="https://bugs.webkit.org/show_bug.cgi?id=269923">https://bugs.webkit.org/show_bug.cgi?id=269923</a>
<a href="https://rdar.apple.com/123119103">rdar://123119103</a>

Reviewed by Simon Fraser.

The PDFPluginChoiceAnnotation sets its font size
based on the size of the font in the underlying
annotation within the document and then scales it
based on its scale factor. The UnifiedPDFPlugin&apos;s
scale factor is its zoom scale factor, but we
could also have scaled the document to fit in the
available width. PDFPluginTextAnnotations take
into consideration both scale factors so we should
do the same here.

In order to do this more cleanly, I introduced
contentScaleFactor on the plugins. For the legacy
PDF plugin this is just the same as its
scaleFactor, but for the UnifiedPDFPlugin this is
both the document fitting scale and the zoom
scale (it seems like this is what the legacy
plugin considers to be its scaleFactor as well).

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::contentScaleFactor const):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginChoiceAnnotation.mm:
(WebKit::PDFPluginChoiceAnnotation::updateGeometry):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginTextAnnotation.mm:
(WebKit::PDFPluginTextAnnotation::updateGeometry):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::contentScaleFactor const):
(WebKit::UnifiedPDFPlugin::updateSnapOffsets):
(WebKit::UnifiedPDFPlugin::searchInDictionary):

Canonical link: <a href="https://commits.webkit.org/275264@main">https://commits.webkit.org/275264@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/77ad5affc838cd9d9cfa3f7ae0594057d012180c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41351 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20365 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43729 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/43915 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37443 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23450 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17695 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/34187 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41925 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17285 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35615 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14848 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/15012 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45253 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37535 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36938 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40675 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/16166 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13263 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/39073 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17785 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9271 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17838 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/17429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -859,6 +859,11 @@ static String mutationObserverNotificationString()
 
+CGFloat UnifiedPDFPlugin::contentScaleFactor() const
+{
+    return m_scaleFactor * m_documentLayout.scale();
```
Looks like we have at least 3 uses of this product inside `UnifiedPDFPlugin.mm`. Would you like to replace those with `contentScaleFactor()` instead?

I see uses in `centeringOffset()`, `updateSnapOffsets()`, and `searchInDictionary()`.

sammygill replied:
I think the logic in centeringOffset was slightly different in that it didn't allow just a simple swap in, but I was able to do the other 2

aprotyas commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -859,6 +859,11 @@ static String mutationObserverNotificationString()
     return m_scaleFactor;
 }
 
+CGFloat UnifiedPDFPlugin::contentScaleFactor() const
```
Nit: Just return `float` here?

smfr replied:
Agree.

aprotyas reviewed (commented):
LGTM!

sammygill committed:
[UnifiedPDF] PDFPluginChoiceAnnotations have incorrect font size.
https://bugs.webkit.org/show_bug.cgi?id=269923
rdar://123119103

Reviewed by Simon Fraser.

The PDFPluginChoiceAnnotation sets its font size
based on the size of the font in the underlying
annotation within the document and then scales it
based on its scale factor. The UnifiedPDFPlugin's
scale factor is its zoom scale factor, but we
could also have scaled the document to fit in the
available width. PDFPluginTextAnnotations take
into consideration both scale factors so we should
do the same here.

In order to do this more cleanly, I introduced
contentScaleFactor on the plugins. For the legacy
PDF plugin this is just the same as its
scaleFactor, but for the UnifiedPDFPlugin this is
both the document fitting scale and the zoom
scale (it seems like this is what the legacy
plugin considers to be its scaleFactor as well).

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::contentScaleFactor const):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginChoiceAnnotation.mm:
(WebKit::PDFPluginChoiceAnnotation::updateGeometry):
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginTextAnnotation.mm:
(WebKit::PDFPluginTextAnnotation::updateGeometry):
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::contentScaleFactor const):
(WebKit::UnifiedPDFPlugin::updateSnapOffsets):
(WebKit::UnifiedPDFPlugin::searchInDictionary):

Canonical link: https://commits.webkit.org/275264@main
